projects/Mallard/src/components/layout/animators/fader.tsx

```diff
@@ -1,43 +1,57 @@
-import React from 'react'
+import React, { useRef } from 'react'
 import {
     getNavigationPosition,
     SaveableNavigationPositions,
 } from 'src/helpers/positions'
-import { Animated, StyleSheet } from 'react-native'
+import { Animated, StyleSheet, View, Dimensions } from 'react-native'
 
 /*
 This is part of the transition from articles to fronts
 and it fades content in and out with a user chosen delay.
 
-The build order is an int X-1 on purpose to abstract the details
-of how and when to fade to this component. We will likely
-wanna mess with the actual animation numbers later on to make them
-more/less staggered
+The build order goes up/down according to screen position
 */
 
 export interface PropTypes {
-    buildOrder: number
-    children: Element
+    first?: boolean
+    children?: Element
     position: SaveableNavigationPositions
 }
 
 const faderStyles = StyleSheet.create({
     wrapper: { width: '100%' },
 })
 
-const fadeOffset = 0.5
+const clamp = (number: number, min: number, max: number) => {
+    if (number > max) return max
+    if (number < min) return min
+    return number
+}
 
-const Fader = ({ buildOrder, children, position }: PropTypes) => {
+const Fader = ({ children, position }: PropTypes) => {
     const navigationPosition = getNavigationPosition(position)
-    if (buildOrder > 6) buildOrder = 6
+    const buildOrder = useRef(0)
+    const faderRef = useRef<View>()
+
     return (
         <Animated.View
+            ref={(view: { _component: unknown }) => {
+                if (view && view._component)
+                    faderRef.current = view._component as View
+            }}
+            onLayout={() => {
+                faderRef.current &&
+                    faderRef.current.measureInWindow((x, y) => {
+                        buildOrder.current =
+                            (y / Dimensions.get('window').height) * 6
+                    })
+            }}
             style={[
                 navigationPosition && {
                     opacity: navigationPosition.position.interpolate({
                         inputRange: [
-                            0.2 + buildOrder / 10,
-                            0.4 + buildOrder / 10,
+                            clamp(0.2 + buildOrder.current / 10, 0, 1),
```

Comment:
Again, there are quite a few numbers here. Might be worth adding a comment saying what the min / max expected input range is here.

Reply:
the specifics are sort of a design decision but basically, you don't wanna have the animation not be a growing set of numbers because it'll throw! will comment

Reply:
Yeah, you're right - I don't think you need to change the code but maybe just a couple of lines in the comment would make it a bit easier to know where they came from 👌 - thanks!